### DISCUSSION TITLE
Network connection loss during active webrepl pauses ESP32 script

### DISCUSSION THREAD

**Discussion by astirfreak** [micropython/micropython#10444] [ESP32]
Hello,
I have a script which is working in a 0.5s task. Within this task I use a lot of print instructions to write the current status of the script to the console (serial and/or webrepl). This works fine. The task itself lasts about 100ms and the print routine, called within the task another about 150ms. So the task has a total time of about 250ms. The remaining time of the 500ms task, the script is in a while loop containing utime.sleep_ms(11) instructions.
I observed, that the duration of the print routine is dependent on the quality of the network connection, if webrepl is connected.

When I have an active webrepl session on my PC and I pull the network cable during this active session and put it in again after several seconds I observe, that the complete script has stopped for the time, the network connection was lost. After reconnect, the script continues without error.

Is there a way to avoid this behavior? I guess, that the delay happens within a print instruction.

Michael

**Comment by astirfreak** [comment]
Meanwhile I solved the problem by putting all prints in an independent thread.
But it's strange, that a print seems to have absolutely no timeout, if the webrepl connection gets lost. So my print-thread will hang forever in that case and only comes back if a new webrepl connection is initiated.

**Reply by jimmo** [reply]
Yes this appears to be a limitation in the implementation of webrepl -- it creates a blocking websocket with no timeout.

I agree this should probably be fixed -- there should be a (short-ish) timeout and the webrepl should un-dupterm itself.

**Reply by jimmo** [reply]
See https://github.com/micropython/micropython/issues/10508
